### DISCUSSION TITLE
Best way to access invoked child machine state in react

### DISCUSSION THREAD

**Discussion by Kashif7** [statelyai/xstate#2202] [Q&A] [answered]
How can I access the state of an invoked child machine using the "useState" hook from a react component?

**Comment by davidkpiano** [comment, marked as answer]
It's not straightforward right now, but there's a pending PR for this: https://github.com/davidkpiano/xstate/pull/2197

**Reply by Kashif7** [reply]
Noted, thanks. Till that I will use spawn to keep a ref for the child machine in the context and accessing the state through that.
